app/code/Magento/Ui/view/base/web/js/grid/columns/multiselect.js

```diff
@@ -448,7 +457,7 @@ define([
                 this.selectAll();
             }
 
-            this.allSelected(allSelected);
+            this.allSelected(pageTotal === pageSelected);
```

Comment:
There are a lot of jasmine test failures could you please take a look ? i think this may cause an issue